src/Solver/DantzigWolfe/DwBundleDual.cpp

```diff
@@ -660,8 +665,25 @@ DSP_RTN_CODE DwBundleDual::addRows(
 		cutvec.clear();
 
 		/** original constraints */
-		if (statuses[s] != DSP_STAT_DUAL_INFEASIBLE)
+		if (statuses[s] != DSP_STAT_DUAL_INFEASIBLE && dualsol_[sind] > objs[s] + 1.0e-8) {
 			cutvec.insert(sind, 1.0);
+			DSPdebugMessage("obj coefficient s: %d, sind: %d, coeff: %f\n", s, sind, osi_->si_->getObjCoefficients()[sind]);
+			// if (osi_->si_->getObjCoefficients()[sind] == 0.0) {
+			// 	osi_->si_->setObjCoeff(sind, -1.0);
+			// 	// sind or s ???
+			// 	DSPdebugMessage("updated obj coefficient s: %d, sind: %d, coeff: %f\n", s, sind, osi_->si_->getObjCoefficients()[sind]);
+			// }
+			if (osi_->si_->getColLower()[sind] == 0.0 && osi_->si_->getColUpper()[sind] == 0.0) {
```

Comment:
Perhaps there is a better way to check whether the bound has been updated or not.

Reply:
create variables that raises flags